includes/functions.php

```diff
@@ -1593,3 +1595,48 @@ function is_self_ping( $id ) {
 
 	return false;
 }
+
+/**
+ * Add an object to the outbox.
+ *
+ * @param mixed   $data               The object to add to the outbox.
+ * @param string  $type               The type of the Activity.
+ * @param integer $user_id            The User-ID.
+ * @param string  $content_visibility The visibility of the content.
+ *
+ * @return boolean|int The ID of the outbox item or false on failure.
+ */
+function add_to_outbox( $data, $type = 'Create', $user_id = 0, $content_visibility = ACTIVITYPUB_CONTENT_VISIBILITY_PUBLIC ) {
+	$transformer = Transformer_Factory::get_transformer( $data );
+
+	if ( ! $transformer || is_wp_error( $transformer ) ) {
+		return false;
+	}
+
+	$activity = $transformer->to_object();
+
+	if ( ! $activity || is_wp_error( $activity ) ) {
+		return false;
+	}
+
+	set_wp_object_state( $data, 'federate' );
+
+	$id = Outbox::add( $activity, $type, $user_id, $content_visibility );
```

Comment:
It seems to me that all of this functionality in `add_to_outbox` should just live inside of `Outbox::add`? Which would then either leave us with a thin wrapper in this function, or we can then get rid of the wrapper entirely?